### PR TITLE
ruby_3_0: drop

### PR DESCRIPTION
**Pull request by RaitoBezarius**
## Description of changes

Ruby 3.0 will be removed as its EOL window is during NixOS 23.11 release.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by erictapen** [comment]
This update to Mastodon contains the change to ruby 3.2: https://github.com/NixOS/nixpkgs/pull/251950

**Review by SuperSandro2000 — APPROVED** [review]
🎉

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 264361` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>gitlab</li>
    <li>gitlab-ee</li>
    <li>mailcatcher</li>
    <li>unit</li>
  </ul>
</details>

File: pkgs/applications/version-management/gitlab/default.nix [hunk 1]
```diff
@@ -17,7 +17,7 @@ let
 
   rubyEnv = bundlerEnv rec {
     name = "gitlab-env-${version}";
-    ruby = ruby_3_0;
```
Comment (dasJ) [inline pkgs/applications/version-management/gitlab/default.nix:20]: But afaik, GitLab isn't officially compatible with 3.2?

- 3.2: https://gitlab.com/gitlab-org/gitlab-development-kit/-/issues/1731
- 3.1 (which works): https://gitlab.com/groups/gitlab-org/-/epics/10034

Comment (RaitoBezarius) [reply pkgs/applications/version-management/gitlab/default.nix:20]: Did I upgrade it to 3.2 by mistake? Then, we should go back to 3.1, sorry for that.

**Pull request merged**